agent/agent.go

```diff
@@ -603,14 +605,48 @@ func (a *Agent) finishTask(ctx context.Context, tc *taskContext, status string,
 
 func (a *Agent) endTaskResponse(tc *taskContext, status string, message string) *apimodels.TaskEndDetail {
 	var description string
-	var cmdType string
+	var failureType string
+	if a.endTaskResp != nil { // if the user indicated a task response, use this instead
+		tc.logger.Task().Infof("Task status set using /task_status")
+		if !evergreen.IsValidTaskEndStatus(a.endTaskResp.Status) {
+			tc.logger.Task().Errorf("'%s' is not a valid task status", a.endTaskResp.Status)
+			status = evergreen.TaskFailed
+			failureType = evergreen.CommandTypeSystem
+			tc.logger.Task().Debugf("END TASK RESP: %v", a.endTaskResp)
```

Comment:
This is a pretty big struct to log, and I'm not sure it's useful. I think logging `a.endTaskResp.Status` on 612 is sufficient.

Reply:
Sorry this was leftover debugging for myself